src/test/java/pl/wedding/management/model/employee/EmployeeRowFactoryTest.java

```diff
@@ -0,0 +1,86 @@
+            .typeOfWork(160)
+            .build();
+
+    EmployeeDto employeeDto = EmployeeDto.builder()
+            .name("Adam")
+            .surname("Kowalski")
+            .workPlace(Object.class)
+            .position(PositionEmployee.CLEANER)
+            .salary(salary)
+            .statusEmployee(StatusEmployee.WORKING)
+            .typeOfContract(TypeOfContract.CONTRACT_OF_EMPLOYMENT)
+            .build();
+
+    @Test
+    @DisplayName("should converter from dto to entity correctly")
+    void converterCorrectly() {
+        //Given
+        EmployeeRowFactory employeeRowFactory = new EmployeeRowFactory();
+        //When
+        var employeeAfterConvert = employeeRowFactory.converter(employeeDto);
+        //Then
+        assertThat(employeeAfterConvert).isInstanceOf(EmployeeEntity.class);
+    }
+
+    @DisplayName("should converter from dto to entity correctly, check field by field")
+    @Test
+    void converterCorrectlyByField(){
+        //Given
+        EmployeeEntity employeeEntity = new EmployeeEntity();
+        employeeEntity.setName("Adam");
+        employeeEntity.setSurname("Kowalski");
+        employeeEntity.setWorkPlace(Object.class);
+        employeeEntity.setPosition(PositionEmployee.CLEANER);
+        employeeEntity.setSalary(salary);
+        employeeEntity.setStatusEmployee(StatusEmployee.WORKING);
+        employeeEntity.setTypeOfContract(TypeOfContract.CONTRACT_OF_EMPLOYMENT);
+        EmployeeRowFactory employeeRowFactory = new EmployeeRowFactory();
+
+        //When
+        var employeeAfterConvert = employeeRowFactory.converter(employeeDto);
+
+        //Then
+        assertEquals(employeeEntity.getName(), employeeAfterConvert.getName());
+        assertEquals(employeeEntity.getSurname(), employeeAfterConvert.getSurname());
+        assertEquals(employeeEntity.getWorkPlace(), employeeAfterConvert.getWorkPlace());
+        assertEquals(employeeEntity.getPosition(), employeeAfterConvert.getPosition());
+        assertEquals(employeeEntity.getSalary(), employeeAfterConvert.getSalary());
+        assertEquals(employeeEntity.getStatusEmployee(), employeeAfterConvert.getStatusEmployee());
+        assertEquals(employeeEntity.getTypeOfContract(), employeeAfterConvert.getTypeOfContract());
+
+    }
+
+
+    @DisplayName("Should throw an NullPointException")
+    @Test
+    void convertNotCorrectly(){
+        //Given
+        EmployeeDto employeeDtoInMethod = EmployeeDto.builder()
+                .name("Adam")
+                .surname("Kowalski")
+                .workPlace(Object.class)
+                .position(PositionEmployee.CLEANER)
+                .typeOfContract(TypeOfContract.CONTRACT_OF_EMPLOYMENT)
+                .build();
+        EmployeeRowFactory employeeRowFactory = new EmployeeRowFactory();
+        //When
+        var employeeAfterConvert = employeeRowFactory.converter(employeeDtoInMethod);
+        //Then
+        assertThatNullPointerException();
```

Comment:
Czemu rzucamy npe?

Reply:
ponieważ brakuje z dto wszystkich danych aby wpisać pracownika w entity, aby dalej z nim coś robić (np zapisać w bazie danych)

Reply:
Zróbmy na to dedykowany wyjątek. NPE jest błędem programistycznym, bardzo często występującym. Widząc NPE w logach, od razu myślisz 'zjebał', a nie że nie ma wymaganych danych. Najlepiej byłoby rozszerzyć `RuntimeException`.
Walidacja czy mamy wszystko co potrzeba powinna być zaraz przy kontrolerze rest, co oczywiście nie oznacza, że konwerter nie może rzucić czymś jak w tym miejscu nie ma wszystkiego co potrzeba.

Reply:
zmieniłem test, sprawdzam, czy rzuca runtimeexception